### ISSUE TITLE
Feature Request: Auto-detect parent branch from GitHub API on first run of sync

### ISSUE THREAD

**Issue by cirego** [git-town/git-town#1297]
First off -- thanks so much for this awesome utility! I cannot say enough good things about it -- it's made a huge difference in my workflow. In particular, the feature where it auto-updates the base branch on child PRs when a parent branch is shipped: :boom:

I've recently started pairing `git-town` with `gh`, the GitHub CLI. This lets me pull down a pull request as a local branch using `gh pr checkout PRID`. When I first run `git town sync`, it asks me for the parent branch. It would be nice if there was a way to auto-detect the parent branch using the GitHub API when there is a Pull Request associated with the current branch.

**Comment by kevgo** [comment]
Hi @cirego thanks and that's a pretty good idea! Listing the pull requests via the [GitHub REST API](https://developer.github.com/v3/pulls/#list-pull-requests) API should be pretty straightforward for this. It allows filtering the pull requests by the `head` branch. Here is an example that finds the pull request where I relax the Ruby version:

```
curl "https://api.github.com/repos/git-town/git-town/pulls?state=all&head=git-town:kg-ruby"
```

**Comment by kevgo** [comment]
Storing the parent of feature branches on GitHub and using that as a sync database would be the missing piece to implement #1030. Brilliant idea!

**Comment by zeronacer** [comment]
I would love to see this for GitLab!

I'm currently researching tooling for my software team and came across this amazing tool which would make code/diff-stacking so much easier. Not being able to sync inheritance trees between team members is a huge deal-breaker though.

**Comment by kevgo** [comment]
Consolidating this ticket into #2745.

**Comment by kevgo** [comment]
This is now available in [v16.3](https://github.com/git-town/git-town/releases/tag/v16.3.0).
